client/android/WorldScope/app/src/main/java/fragment/CommentFragment.java

```diff
@@ -119,7 +123,8 @@ private void setUpButton() {
         OnClickListener btnListener = new OnClickListener() {
             public void onClick(View v) {
 
-                if(commentEditText.getText().toString() != null) {
+                if(commentEditText.getText().toString() != null && commentEditText.getText().toString().length() != 0) {
+                    Log.d(TAG, "Sending: " + commentEditText.getText().toString());
```

Comment:
Should the check be for `commentEditText.getText() != null` instead?
I don't quite know how the class works, but it seems odd if the `toString` returns a null but the `getText` doesn't.

Also, rather than `.length() != 0`, you probably want to just do a `!.isEmpty()`.